### ISSUE TITLE
save then load gives less accurate results

### ISSUE THREAD

**Issue by drujensen** [NeuraLegion/shainet#48]
I have created a model for the Kaggle titanic competition.  I ran into an issue related to saving and loading from files generates different results.

The first thing I wanted to try was to create an NN that will predict the ages of the passengers that they didn't know the age of.  Before I saved the model, I was getting decent results.  The ages were fairly close.  This accuracy was based on exact age matches but I also manually verified that the ages were close.
```
Training size: 714
----------------------
T: 55 | F: 659
----------------------
Accuracy: 0.07703081232492998
```

After saving and loading the model and then running on the same test data, I am getting less accurate results and the ages are not close.  They all come back as just a subset of possible ages.
```
Training size: 714
----------------------
T: 21 | F: 693
----------------------
Accuracy: 0.029411764705882353
```

**Comment by drujensen** [comment]
```
dependencies:
  shainet:
    github: NeuraLegion/shainet
    branch: master
```
```
$shards update
Fetching https://github.com/NeuraLegion/shainet.git
Fetching https://github.com/Papierkorb/cannon.git
Installing shainet (0.1.0 at master)
Using cannon (0.2.1)
```

**Comment by drujensen** [comment]
I will look at the specs and make sure I am doing this correctly

**Comment by drujensen** [comment]
@bararchy I created a PR to demonstrate the predicted results are different after loading from a file

https://github.com/NeuraLegion/shainet/pull/50

**Comment by drujensen** [comment]
@bararchy i did some testing.  If you load and immediate save a network and then compare the two files, you will see the differences:
```
require "./src/shainet.cr"

training_data = [
  [[0, 0], [0]],
  [[1, 0], [1]],
  [[0, 1], [1]],
  [[1, 1], [0]],
]

xor : SHAInet::Network = SHAInet::Network.new
xor.add_layer(:input, 2, :memory, SHAInet.sigmoid)
1.times { |x| xor.add_layer(:hidden, 3, :memory, SHAInet.sigmoid) }
xor.add_layer(:output, 1, :memory, SHAInet.sigmoid)
xor.fully_connect
# data, training_type, cost_function, activation_function, epochs, error_threshold (MSE %), log each steps
xor.train(training_data, :sgdm, :mse, epochs = 5000, threshold = 0.000001, log = 100)
xor.save_to_file("./xor.nn")

xor2 : SHAInet::Network = SHAInet::Network.new
xor2.load_from_file("./xor.nn")

xor2.save_to_file("./xor2.nn")
```

`xor2` is larger and the `input` neurons have `synapses_in` where the `xor` the `synapses_in` is empty.  I think its supposed to be empty since this is the first layer.

**Comment by bararchy** [comment]
Good debugging, nice work , let me go over it and test

**Comment by drujensen** [comment]
I think I tracked it down.  i will submit a PR

**Comment by drujensen** [comment]
@bararchy Here is the fix.  We were adding the synapse to the wrong neurons.  https://github.com/NeuraLegion/shainet/pull/50

**Comment by drujensen** [comment]
fixed by #50
